lib/adal.js

```diff
@@ -189,6 +200,28 @@ var AuthenticationContext = (function () {
         // callback from redirected page will receive fragment. It needs to call oauth2Callback
     };
 
+    AuthenticationContext.prototype.loginPopup = function (urlNavigate) {
+        var popupWindow = window.open(urlNavigate, "login", 'width=483, height=600');
+        if (popupWindow && popupWindow.focus)
+            popupWindow.focus();
+        var that = this;
+        var registeredRedirectUri = this.config.redirectUri;
+        var pollTimer = window.setInterval(function () {
+            if (!popupWindow || popupWindow.closed || popupWindow.closed === undefined) {
+                window.clearInterval(pollTimer);
+            }
+            try {
+                if (popupWindow.document.URL.indexOf(registeredRedirectUri) != -1) {
```

Comment:
Why check for the redirectUri in the popup window URL? Is there an attack we're preventing?